components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -213,21 +279,34 @@ GetTransactionInfoFromData(const std::vector<uint8_t>& data) {
     //
     // Ref:
     // https://github.com/0xProject/protocol/blob/b46eeadc64485288add5940a210e1a7d0bcb5481/contracts/zero-ex/contracts/src/features/interfaces/ITransformERC20Feature.sol#L113-L134
-    auto decoded_calldata = ABIDecode(
-        {"address", "address", "uint256", "uint256", "(uint32,bytes)[]"},
-        calldata);
+    auto type = eth_abi::Tuple()
+                    .addTupleType(eth_abi::Address())
+                    .addTupleType(eth_abi::Address())
+                    .addTupleType(eth_abi::UintM(256))
+                    .addTupleType(eth_abi::UintM(256))
+                    .addTupleType(
+                        eth_abi::Array()
+                            .setArrayType(eth_abi::Tuple()
+                                              .addTupleType(eth_abi::UintM(32))
+                                              .addTupleType(eth_abi::Bytes())
+                                              .build())
+                            .build())
+                    .build();
+    auto decoded_calldata = ABIDecode(type, calldata);
     if (!decoded_calldata) {
       return std::nullopt;
     }
 
-    const auto& tx_args = std::get<1>(*decoded_calldata);
     return std::make_tuple(
         mojom::TransactionType::ETHSwap,
         std::vector<std::string>{"bytes",     // fill path,
                                  "uint256",   // maker amount
                                  "uint256"},  // taker amount
-        std::vector<std::string>{tx_args.at(0) + tx_args.at(1).substr(2),
-                                 tx_args.at(2), tx_args.at(3)});
+        std::vector<std::string>{
```

Comment:
doesn't match with `type` at all

Reply:
Btw this code has not been changed. We do not pass the exact parsed calldata in order to simply the frontend integration - they just need to know how to handle `ETHSwap` transactions.